### PR TITLE
[ISSUE-717] Generic ephemeral volumes support

### LINKED ISSUE #717: Generic ephemeral volumes support
**Is your feature request related to a problem? Please describe.**
Need to support Ephemeral volumes in CSI. Changes must be done in Scheduler Extender to handle EphemeralVolumeSource case.

**Describe the solution you'd like**
Code to be updated - https://github.com/dell/csi-baremetal/blob/16a14ae575d7273d13f851b13a212be5832165b7/pkg/scheduler/extender/extender.go#L222

### PR DESCRIPTION
**Pull request by safronovD**
### Resolves #717 

- Add ephemeral volume selecting in extender
- Add example

## PR checklist
- [x] Add link to the issue
- [x] Choose Project
- [x] Choose PR label
- [x] New unit tests added
- [ ] Modified code has meaningful comments
- [ ] All TODOs are linked with the issues
- [ ] All comments are resolved

## Testing
_Provide test details_

### CODE REVIEW

File: pkg/scheduler/extender/extender.go [hunk 1]
```diff
@@ -594,3 +609,7 @@ func (e *Extender) scNameStorageTypeMapping(ctx context.Context) (map[string]str
 	}
 	return scNameTypeMap, nil
 }
+
+func generateEphemeralVolumeName(podName, volumeName string) string {
```
Comment (rkiselenko) [inline pkg/scheduler/extender/extender.go:613]: What the point of separate function ? Just use `fmt.Sprintf("%v-%v", podName, volume.Name)`

**Review by rkiselenko — APPROVED** [review]

File: pkg/scheduler/extender/extender.go [hunk 2]
```diff
@@ -250,6 +250,21 @@ func (e *Extender) gatherCapacityRequestsByProvisioner(ctx context.Context, pod
 			}
 			continue
 		}
+		if v.Ephemeral != nil {
+			claimSpec := v.Ephemeral.VolumeClaimTemplate.Spec
+			if storageType, ok := scs[*claimSpec.StorageClassName]; ok {
+				storageReq, ok := claimSpec.Resources.Requests[coreV1.ResourceStorage]
+				if !ok {
+					ll.Errorf("There is no key for storage resource for PVC %s", v.Name)
```
Comment (mishoyama) [inline pkg/scheduler/extender/extender.go:258]: Does it mean that incorrect SC name was passed?

**Review by mishoyama — COMMENTED** [review]
Please update documentation - supported features list

File: pkg/scheduler/extender/extender.go (hunk 1 above)
Comment (safronovD) [reply pkg/scheduler/extender/extender.go:613]: We should have the same names during both ephemeral and generic ephemeral scenario. And for me, `requestName := generateEphemeralVolumeName(podName, volume.Name)` looks better, it marks that the whole name is not just a Volume name

File: pkg/scheduler/extender/extender.go (hunk 2 above)
Comment (safronovD) [reply pkg/scheduler/extender/extender.go:258]: No, it means that Volume size is not set, but from k8s specification it always must not be empty. Just check to not raise panic here, if we received incorrect data

**Comment by safronovD** [comment]
> Please update documentation - supported features list

updated

**Review by mishoyama — APPROVED** [review]

File: pkg/scheduler/extender/extender.go (hunk 1 above)
Comment (rkiselenko) [reply pkg/scheduler/extender/extender.go:613]: But it implies from the code, this code tell you that the string is `podName - volume.Name` , there is no need of separate function for string formatting.

Comment (safronovD) [reply pkg/scheduler/extender/extender.go:613]: In my opinion, it's not clear from the code, why we don't use VolumeName as it works with PVC. I've spent some time when just putting `v.Name` here and then decided to create a specific function with `EphemeralVolumes` in signature

Comment (safronovD) [reply pkg/scheduler/extender/extender.go:613]: Go compiler should inline such simple functions, so it's just formatting question

**Pull request merged**